truetype/truetype.go

```diff
@@ -345,6 +433,20 @@ func (f *Font) Index(x rune) Index {
 	return 0
 }
 
+// Name returns the NameID value of a Font.
+// Returns "" on error or not found.
+func (f *Font) Name(id NameID) string {
+	valueOffset, valueLength, pid, _ := parseSubtables(f.name, "name", 6, 12, 10, 8, []int{6}, []int{int(id)}, true)
+	if valueOffset == 0 {
+		return ""
+	}
+
+	stringOffset := int(u16(f.name, 4))
+	// Return the ASCII value of the encoded string.
+	// The string is encoded as UTF-16 on non-Apple pids; Apple is pid 1.
+	return nameEntryInASCII(f.name[stringOffset+valueOffset:stringOffset+valueOffset+valueLength], pid != 1)
```

Comment:
I'd just inline nameEntryInASCII here.

	src := f.name[offset:offset + length]
	var dst []byte
	if pid != 1 { // UTF-16.
		if len(src)&1 != 0 {
			return ""
		}
		dst = make([]byte, 0, len(src) / 2)
		for i := 0; i < len(src); i += 2 {
			dst = append(dst, printable(u16(src, i)))
		}
	} else { // ASCII.
		dst = make([]byte, 0, len(src))
		for _, c := range src {
			dst = append(dst, printable(uint16(c)))
		}
	}
	return string(dst)
}

func printable(r uint16) byte {
	if 0x20 <= r && r <= 0x7f {
		return byte(r)
	}
	return '?'
}